core/src/main/java/org/elasticsearch/action/support/replication/ReplicationRequest.java

```diff
@@ -171,12 +172,12 @@ long routedBasedOnClusterVersion() {
     }
 
     /** returns the primary term active at the time the operation was performed on the primary shard */
-    public long primaryTerm() {
+    long primaryTerm() {
         return primaryTerm;
```

Comment:
we cannot assert "non-0" here, as we are calling the method when deserializing a ConcreteShardRequest which comes from a < 5.6 node. Such a node will not be sending a "non-0" value for the primary action, hence the current special handling in `PrimaryOperationTransportHandler`